File: all-submissions/1320-remove-all-adjacent-duplicates-in-string-ii/solution.java
class Solution {
    public String removeDuplicates(String s, int k) {
        Stack<Pair<Character, Integer>> stack = new Stack<>();
        if(s.length() == 0) return s;
        for(char c: s.toCharArray()) {
            if(!stack.isEmpty()) {
                Pair<Character, Integer> curr = stack.peek();
                if (curr.getKey() == c) {
                    int val = curr.getValue()+1;
                    if(val == k) {
                        while(!stack.isEmpty() && stack.peek().getKey() == c) {
                            stack.pop();
                        }
                        continue;
                    } else {
                        stack.push(new Pair(c, val));
                    }
                } else {
                    stack.push(new Pair(c, 1));
                }
            } else {
                stack.push(new Pair(c, 1));
            }
        }
        StringBuilder sb = new StringBuilder();
        while(!stack.isEmpty()) {
            sb.append(stack.pop().getKey());
        }
        return sb.reverse().toString();
        
    }
}

File: all-submissions/1320-remove-all-adjacent-duplicates-in-string-ii/solution.py
class Solution:
    def removeDuplicates(self, s: str, k: int) -> str:
        stack = []
        for ch in s:
            if not stack or stack[-1][0] != ch:
                stack.append([ch,1])
            elif stack[-1][0] == ch:
                count = stack[-1][1] + 1
                stack.append([ch, count])
                if count == k:
                    while count > 0:
                        stack.pop()
                        count -=1
        res = ""
        for ch, count in stack:
            res += ch

        return res
        
